### PR TITLE
update polling to 3.x

### PR DESCRIPTION
**Pull request by Fabian-Gruenbichler**
didn't do any in-depth tests and don't know this code base all to well, so please don't hesitate to do things differently as you see fit.

Debian has recently upgrade to polling 3.x, and this is one of the few packaged crates still using the 2.x release series..

### CODE REVIEW

File: src/service_daemon.rs [hunk 1]
```diff
@@ -482,10 +482,13 @@ impl ServiceDaemon {
     /// 5. process retransmissions if any.
     fn run(mut zc: Zeroconf, receiver: Receiver<Command>) -> Option<Command> {
         // Add the daemon's signal socket to the poller.
-        if let Err(e) = zc.poller.add(
-            &zc.signal_sock,
-            polling::Event::readable(SIGNAL_SOCK_EVENT_KEY),
-        ) {
+        #[allow(unsafe_code)]
```
Comment (keepsimple1) [inline src/service_daemon.rs:485]: Thanks for your PR! I want to keep this library safe code only, unless we really have to break it. 

Let me take a look at our options.

**Comment by keepsimple1** [comment]
As I wanted to keep this lib safe code only, and `polling` 3.x does not provide a safe API (`add` is unsafe),  I'm trying to move away from `polling` and use `mio` instead (see PR #280 ).  

Could you please confirm if `mio` 1.0 is allowed / packaged in Debian? Thanks!

**Comment by Fabian-Gruenbichler** [comment]
it is (in unstable/testing, which will become trixie/stable next year :))

https://tracker.debian.org/pkg/rust-mio

**Comment by keepsimple1** [comment]
Thanks for confirming. I've merged PR #280, which replaced this PR. Will close this PR now. If any related questions, please feel free to open a new issue or PR. Thanks!

**Pull request closed without merging**